### ISSUE TITLE
cudnn PoolBackward launch failed when using tf.nn.max_pool on Tensorflow GPU (Windows 10)

### ISSUE THREAD

**Issue by rktamplayo** [tensorflow/tensorflow#13292]
When using max_pool, the error below shows and stops the code. I used the code available here: [https://github.com/charlesashby/CharLSTM](https://github.com/charlesashby/CharLSTM). Specifically, I used the `lib_model/bidirectional_lstm.py` and the error occurs at the `tdnn` function when `tf.nn.max_pool` is ran.

What does this error mean? Thanks!

```
Caused by op 'gradients/TDNN/MaxPool_grad/MaxPoolGrad', defined at:
  File "main.py", line 33, in <module>
    network.train()
  File "E:\rktamplayo\Personalized\CharLSTM\lib_model\bidirectional_lstm.py", line 165, in train
    optimizer = tf.train.AdamOptimizer(learning_rate=learning_rate).minimize(cost)
  File "C:\Users\seung-won\Anaconda3\envs\tensorflow\lib\site-packages\tensorflow\python\training\optimizer.py", line 315, in minimize
    grad_loss=grad_loss)
  File "C:\Users\seung-won\Anaconda3\envs\tensorflow\lib\site-packages\tensorflow\python\training\optimizer.py", line 386, in compute_gradients
    colocate_gradients_with_ops=colocate_gradients_with_ops)
  File "C:\Users\seung-won\Anaconda3\envs\tensorflow\lib\site-packages\tensorflow\python\ops\gradients_impl.py", line 542, in gradients
    grad_scope, op, func_call, lambda: grad_fn(op, *out_grads))
  File "C:\Users\seung-won\Anaconda3\envs\tensorflow\lib\site-packages\tensorflow\python\ops\gradients_impl.py", line 348, in _MaybeCompile
    return grad_fn()  # Exit early
  File "C:\Users\seung-won\Anaconda3\envs\tensorflow\lib\site-packages\tensorflow\python\ops\gradients_impl.py", line 542, in <lambda>
    grad_scope, op, func_call, lambda: grad_fn(op, *out_grads))
  File "C:\Users\seung-won\Anaconda3\envs\tensorflow\lib\site-packages\tensorflow\python\ops\nn_grad.py", line 526, in _MaxPoolGrad
    data_format=op.get_attr("data_format"))
  File "C:\Users\seung-won\Anaconda3\envs\tensorflow\lib\site-packages\tensorflow\python\ops\gen_nn_ops.py", line 1752, in _max_pool_grad
    data_format=data_format, name=name)
  File "C:\Users\seung-won\Anaconda3\envs\tensorflow\lib\site-packages\tensorflow\python\framework\op_def_library.py", line 767, in apply_op
    op_def=op_def)
  File "C:\Users\seung-won\Anaconda3\envs\tensorflow\lib\site-packages\tensorflow\python\framework\ops.py", line 2630, in create_op
    original_op=self._default_original_op, op_def=op_def)
  File "C:\Users\seung-won\Anaconda3\envs\tensorflow\lib\site-packages\tensorflow\python\framework\ops.py", line 1204, in __init__
    self._traceback = self._graph._extract_stack()  # pylint: disable=protected-access

...which was originally created as op 'TDNN/MaxPool', defined at:
  File "main.py", line 31, in <module>
    network.build()
  File "E:\rktamplayo\Personalized\CharLSTM\lib_model\bidirectional_lstm.py", line 103, in build
    cnn = tdnn(self.X, kernels, kernel_features)
  File "E:\rktamplayo\Personalized\CharLSTM\lib_model\bidirectional_lstm.py", line 92, in tdnn
    pool = tf.nn.max_pool(tf.nn.tanh(conv), [1, 1, reduced_length, 1], [1, 1, 1, 1], 'VALID')
  File "C:\Users\seung-won\Anaconda3\envs\tensorflow\lib\site-packages\tensorflow\python\ops\nn_ops.py", line 1772, in max_pool
    name=name)
  File "C:\Users\seung-won\Anaconda3\envs\tensorflow\lib\site-packages\tensorflow\python\ops\gen_nn_ops.py", line 1605, in _max_pool
    data_format=data_format, name=name)
  File "C:\Users\seung-won\Anaconda3\envs\tensorflow\lib\site-packages\tensorflow\python\framework\op_def_library.py", line 767, in apply_op
    op_def=op_def)
  File "C:\Users\seung-won\Anaconda3\envs\tensorflow\lib\site-packages\tensorflow\python\framework\ops.py", line 2630, in create_op
    original_op=self._default_original_op, op_def=op_def)
  File "C:\Users\seung-won\Anaconda3\envs\tensorflow\lib\site-packages\tensorflow\python\framework\ops.py", line 1204, in __init__
    self._traceback = self._graph._extract_stack()  # pylint: disable=protected-access

InternalError (see above for traceback): cudnn PoolBackward launch failed
         [[Node: gradients/TDNN/MaxPool_grad/MaxPoolGrad = MaxPoolGrad[T=DT_FLOAT, data_format="NHWC", ksize=[1, 1, 16, 1], padding="VALID", strides=[1, 1, 1, 1], _device="/job:localhost/replica:0/task:0/gpu:0"](TDNN/Tanh, TDNN/MaxPool, gradients/TDNN/Squeeze_grad/Reshape)]]
```

**What related GitHub issues or StackOverflow threads have you found by searching the web for your problem?**

I have searched through all websites including GitHub and StackOverflow, and it seems like this error has not been reported anywhere else.

**Environment info**

- Tensorflow 1.3.0
- Python 3.5.3
- CUDA 8.0
- cuDNN 6.0
- OS: Windows 10
- GPU: GeForce GTX 1080ti

**Comment by nikhilpriyatam** [comment]
I got the same issue. By any chance, did you find out a solution to this error?

**Comment by rktamplayo** [comment]
I found that decreasing the number of parameters per run (e.g. by decreasing the batch size) solves the error.

**Comment by drpngx** [comment]
OK, closing for now, since there is an (imperfect) workaround.

**Comment by svjan5** [comment]
I am also facing the same issue. Please let us know a proper solution to this problem.

**Comment by bzamecnik** [comment]
It can happen when some zombie process (invisible in nvidia-smi) has some memory allocated on the GPU.

**Comment by ShiraStarL** [comment]
I had the same problem
Solve it like this:

Check CUDA version with
nvidia-smi
Result was 11.0

Check CUDA version using
nvcc --version
Result was 10.0

cd usr/local
rm cuda
ln -s cuda-11.0 cuda

The problem was that the link wasn't correct.

**Comment by aaniin** [comment]
I faced the same problem with `cudnn PoolBackward launch failed` when using a `MaxPooling2D` layer in a network. 
 
Similar to @ShiraStarL (thanks!!) I had a **mismatch in CUDA versions**. I'm running tensorflow 2.4 and at time of installation had properly installed it with CUDA 11.0. Meanwhile I had installed another CUDA version which updated my `Path` variable. 
To solve the issue on Windows 10, I made sure that `Path` points to the correct CUDA 11.0 as specified in https://www.tensorflow.org/install/source_windows#gpu
